src/Marten/Events/Projections/ProjectionOptions.cs

```diff
@@ -177,13 +202,20 @@ public void Add<TProjection>(ProjectionLifecycle? lifecycle = null) where TProje
     /// <param name="projection"></param>
     /// <typeparam name="T"></typeparam>
     /// <param name="lifecycle">Optionally override the ProjectionLifecycle</param>
-    public void Add<T>(GeneratedAggregateProjectionBase<T> projection, ProjectionLifecycle? lifecycle = null)
+    /// <param name="asyncConfiguration">Use it to define behaviour during projection rebuilds</param>
+    public void Add<T>(
+        GeneratedAggregateProjectionBase<T> projection,
+        ProjectionLifecycle? lifecycle = null,
+        Action<AsyncOptions> asyncConfiguration = null
+    )
     {
         if (lifecycle.HasValue)
         {
             projection.Lifecycle = lifecycle.Value;
         }
 
+        asyncConfiguration?.Invoke(projection.Options);
```

Comment:
Do you just want to go all the way for v6 and remove Lifecycle from the projection classes and make users explicitly specify that as part of registration?